drivers/ruby/lib/net.rb

```diff
@@ -647,6 +649,44 @@ def connect()
       self
     end
 
+    def init_socket
+      unless @ssl_opts.empty?
+        @tcp_socket = base_socket
+        @tcp_socket.connect(::Socket.pack_sockaddr_in(@port, @host))
+        context = create_context(@ssl_opts)
+        @socket = OpenSSL::SSL::SSLSocket.new(@tcp_socket, context)
+        @socket.sync_close = true
+        @socket.connect
+        verify_cert!(@socket, context)
+      else
+        @socket = base_socket
+        @socket.connect(Socket.pack_sockaddr_in(@port, @host))
+      end
+    end
+
+    def base_socket
+      socket = Socket.new(Socket::PF_INET, Socket::SOCK_STREAM, 0)
```

Comment:
This used to be `TCPSocket.open`, right?  What was the motivation for the change?  (Also, are there any cases where this behaves differently?)